feature-staking-impl/src/main/java/io/novafoundation/nova/feature_staking_impl/domain/nominationPools/claimRewards/validations/Declarations.kt

```diff
@@ -3,19 +3,39 @@ package io.novafoundation.nova.feature_staking_impl.domain.nominationPools.claim
 import io.novafoundation.nova.common.validation.ValidationSystem
 import io.novafoundation.nova.common.validation.ValidationSystemBuilder
 import io.novafoundation.nova.feature_staking_impl.domain.common.validation.profitableAction
+import io.novafoundation.nova.feature_wallet_api.data.network.blockhain.assets.AssetSourceRegistry
+import io.novafoundation.nova.feature_wallet_api.data.network.blockhain.assets.existentialDeposit
+import io.novafoundation.nova.feature_wallet_api.domain.validation.enoughTotalToStayAboveED
 import io.novafoundation.nova.feature_wallet_api.domain.validation.sufficientBalance
+import io.novafoundation.nova.runtime.ext.utilityAsset
 
 typealias NominationPoolsClaimRewardsValidationSystem =
     ValidationSystem<NominationPoolsClaimRewardsValidationPayload, NominationPoolsClaimRewardsValidationFailure>
+
 typealias NominationPoolsClaimRewardsValidationSystemBuilder =
     ValidationSystemBuilder<NominationPoolsClaimRewardsValidationPayload, NominationPoolsClaimRewardsValidationFailure>
 
-fun ValidationSystem.Companion.nominationPoolsClaimRewards(): NominationPoolsClaimRewardsValidationSystem = ValidationSystem {
+fun ValidationSystem.Companion.nominationPoolsClaimRewards(
+    assetSourceRegistry: AssetSourceRegistry
```

Comment:
I suggest to create a factory for this validation that later can be passed to concrete validation systems. It would prevent us to modify all places where validation is used in case it will require new dependencies in the future